src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,184 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * UI class.
+ * Handles input from user.
+ * Contains task list, storage and parser.
+ *
+ * @author YanCheng
+ */
+public class Ui {
+
+    private TaskList taskList;
+    private Storage storage;
+    private Parser parser;
+
+    private final String HELP_TEXT = "Duke Bot Commands:\n" +
+            "list : list out all current tasks\n" +
+            "find <keyword> : find all task that corresponds to the keyword\n" +
+            "done <task number> : marks the specified task as done\n" +
+            "delete <task number> : deletes the specified task\n" +
+            "todo <task name> : adds a ToDo task\n" +
+            "deadline <task name> /by YYYY-MM-DD : adds a Deadline task\n" +
+            "event <task name> /at YYYY-MM-DD TT:TT-TT:TT : adds an Event task\n" +
+            "Do note that Date and Time must have the specified format\n";
+
+    /**
+     * Constructor for Ui class.
+     * @param taskList A TaskList object
+     * @param storage A Storage object
+     * @param parser A Parser object
+     */
+    public Ui(TaskList taskList, Storage storage, Parser parser) {
+        this.taskList = taskList;
+        this.storage = storage;
+        this.parser = parser;
+    }
+
+    /**
+     * Prompts user for input and delegates input to Parser.
+     * For use with CLI.
+     */
+    public void echo() {
+
+        Scanner sc = new Scanner(System.in);
+
+        // read duke.txt file and initialise taskList
+        try {
+            storage.init();
+        } catch (DukeException e) {
+            System.out.println(e.getMessage());
+        }
+
+        while (true) {
+            String input = sc.nextLine();
+            if (input.equals("bye")) {
+                break;
+            } else if (input.equals("list")) {
+                System.out.println(taskList.listOut());
+            } else if (input.contains("find")) {
+                System.out.println(taskList.find(input));
+            } else if (input.contains("done")) {
+                try {
+                    System.out.println(taskList.done(input));
+                    storage.save();
+                } catch (DukeException e) {
+                    System.out.println("    ____________________________________________________________");

```

Comment:
This would definitely improve the abstraction! On this note, perhaps the method could be used to print a message within the 2 borders? sth like "_______" + message + "_______".